src/actions/analytics.jsx

```diff
@@ -0,0 +1,90 @@
+
+let Analytics = {
+
+  tagManagerIsLoaded: false,
+
+  EVENT_CATEGORY: 'Consumer Complaint Search',
+
+  /**
+   * Get data layer object.
+   * @param {string} action Name of event.
+   * @param {string} label DOM element label.
+   * @param {string} category Type of event.
+   * @param {Function} callback Function to call on GTM submission.
+   * @param {number} timeout Callback invocation fallback time.
+   * @returns {object} Data layer object.
+   */
+  getDataLayerOptions: function( action, label, category, callback, timeout ) {
+    return {
+      event:         category || Analytics.EVENT_CATEGORY,
+      action:        action,
+      label:         label || '',
+      eventCallback: callback,
+      eventTimeout:  timeout || 500
+    }
+  },
+
+  /**
+   * Initialize the Analytics module.
+   */
+  init: function() {
+    // detect if Google tag manager is loaded
+    if ( window.hasOwnProperty( 'google_tag_manager' ) ) {
+      Analytics.tagManagerIsLoaded = true;
+    } else {
+      var _tagManager;
+      Object.defineProperty( window, 'google_tag_manager', {
+        enumerable: true,
+        configurable: true,
+        get: function() {
+          return _tagManager
+        },
+        set: function( value ) {
+          _tagManager = value;
+          Analytics.tagManagerIsLoaded = true;
+        }
+      } )
+    }
+  },
+
+  /**
+   * @name sendEvent
+   * @kind function
+   *
+   * @description
+   * Pushes an event to the GTM dataLayer.
+   * @param {object} dataLayerOptions Type of event.
+   */
+  sendEvent: function( dataLayerOptions ) {
+    console.log('SEND EVENT GA FUNCTION FIRED')
+    var callback = dataLayerOptions.eventCallback;
+    if ( Analytics.tagManagerIsLoaded ) {
+      window.dataLayer.push( dataLayerOptions );
+    } else if ( callback && typeof callback === 'function' ) {
+      callback();  // eslint-disable-line callback-return, inline-comments, max-len
+    }
+  }
+
+  /**
+   * @name sendEvents
+   * @kind function
+   *
+   * @description
+   * Pushes multiple events to the GTM dataLayer.
+   *
+   * @param {array} eventsArray Array of event objects.
+   */
+  // We don't require sending multiple events at this time.
```

Comment:
I'm not crazy about leaving commented-out code in the codebase.  But I do understand the motivation here.

Is there a public repo that has this function already that we could point to?  Something like:

```Javascript
// sendEvents(array) is at http://example.com/foo/master/bar.js
```